### PR TITLE
Adds example apps

### LINKED ISSUE #107: Add example apps
[One-Pager](https://honeycomb.quip.com/kTyJA9ueDiAa/Beestro-Testing-One-Pager)

Add example apps inline with the repo:
1. Spring Boot app instrumented with the agent. The app should exercise at least one auto-instrumentation (e.g. DB or network request), and have at least one manually created span.
2. Spring Boot app instrumented with the SDK. The app should have at least one manually created span (no auto-instrumentation in the SDK path).

Both apps should have a single curl-able endpoint to exercise the above instrumentations.
Both apps should use latest code version of the agent and SDK available in the repo (local dependency as opposed to pulling from maven central)

### PR DESCRIPTION
**Pull request by JamieDanielson**
## Which problem is this PR solving?

- Closes #107 

## Short description of the changes

- This adds example apps - one using agent, one using SDK. It uses local code for dependencies.
- This also updates the .gitignore file to exclude more java files that are auto-created from editors and such

### CODE REVIEW

**Comment by cartermp** [comment]
@jessitron would be good to get your 👀 on this one too!

File: examples/spring-sdk/src/main/java/io/honeycomb/examples/springbootsdk/Application.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+@SpringBootApplication
+public class Application {
+
+	@Bean
+	public HoneycombSdk honeycomb() {
+			return new HoneycombSdk.Builder()
+					.addSpanProcessor(new BaggageSpanProcessor())
+					.setApiKey(System.getenv("HONEYCOMB_API_KEY"))
+					.setDataset(System.getenv("HONEYCOMB_DATASET"))
+					// .setServiceName(System.getenv("SERVICE_NAME"))
```
Comment (MikeGoldsmith) [inline examples/spring-sdk/src/main/java/io/honeycomb/examples/springbootsdk/Application.java:23]: We should remove the commented out version of `setServiceName`

**Review by MikeGoldsmith — APPROVED** [review]
Looks great, just one minor suggestion of removing a duplicate, commented out version of `setServiceName` 👍🏻

File: examples/spring-sdk/src/main/java/io/honeycomb/examples/springbootsdk/Application.java (hunk 1 above)
Comment (JamieDanielson) [reply examples/spring-sdk/src/main/java/io/honeycomb/examples/springbootsdk/Application.java]: Good catch!

**Pull request merged**